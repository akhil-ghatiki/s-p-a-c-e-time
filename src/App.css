.App {
  background-color: #fffffc;
  height: 100vh;
}

.image {
  width: 100vw;
  height: 360px;
}
