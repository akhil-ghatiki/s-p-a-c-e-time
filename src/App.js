import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import './App.css';
import Grid from '@material-ui/core/Grid';
import PrimaryAppBar2 from "./components/PrimaryAppBar";
import Comments from "./components/Comments";
import Typography from '@material-ui/core/Typography';
import Api from "./utils/Api";
import MarkdownMembrane from "./components/MarkdownMembrane";

const useStyles = (theme) => ({
  root: {
    flexGrow: 1,
  },
  paper: {
    padding: theme.spacing(2),
  },
  control: {
    padding: theme.spacing(2),
  },
  commentHeader: {
    margin: 15
  },
  link: {
    fontFamily: '-apple-system, BlinkMacSystemFont,'
      + ' \'Segoe UI\', \'Roboto\', \'Oxygen\','
      + ' \'Ubuntu\', \'Cantarell\', \'Fira Sans\', \'Droid Sans\','
      + ' \'Helvetica Neue\', sans-serif',
  }
});

class App extends Component {
  constructor(props) {
    super(props)
    this.state = { markDownContent: null, commentsData: [] }
  }

  componentDidMount() {
    this.updateCommentsState();
  }

  static getDerivedStateFromProps(){

  }

  updateCommentsState() {
    Api.get('/issues/1/comments')
      .then((response) => {
        this.setState({ commentsData: response.data.map(comment => this.createCommentData(comment)) })
      });
  }

  createCommentData(comment) {
    let commentDetails = {};
    commentDetails['comment'] = comment.body;
    commentDetails['ImgSrc'] = comment.user.avatar_url;
    commentDetails['commentUrl'] = comment.html_url;
    commentDetails['id'] = comment.id;
    commentDetails['userName'] = comment.user.login;
    commentDetails['createdDateTime'] = comment.created_at;
    return commentDetails;
  }

  render() {
    const { classes } = this.props;
    const gitIssueBaseUrl = "https://github.com/akhil-ghatiki/akhil-ghatiki.github.io/issues/";
    const gitIssueUrl = gitIssueBaseUrl.concat(this.props.gitIssue);
    return (
      <div className={'App'}>
        <PrimaryAppBar2></PrimaryAppBar2>
        <Grid container justify="center" className={classes.root}>
          <Grid item xs={12} sm={3.5}>
            <div></div>
          </Grid>
          <Grid item xs={12} sm={5}>
            <div>
              <MarkdownMembrane markDownFilePath={this.props.contentFilePath}>
              </MarkdownMembrane>
            </div>
            <p className={classes.commentHeader}>Try adding a comment in <span>
              <a href={gitIssueUrl}
                target="_blank" rel="noopener noreferrer">this issue</a></span> and it should show up
                in the comments below</p>
            <Typography className={classes.commentHeader} variant={"h6"}>Comments:</Typography>
            {this.state.commentsData.map(commentDataObject =>
              <Comments commentData={commentDataObject} gitIssue={this.props.gitIssue}></Comments>
            )}
          </Grid>
          <Grid item xs={12} sm={3.5}>
            <div></div>
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default withStyles(useStyles)(App);
