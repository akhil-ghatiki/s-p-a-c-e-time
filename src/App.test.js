import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders rage text', () => {
  const { getByText } = render(<App />);
  const textElement = getByText(/Rage, rage against the dying of the light/i);
  expect(textElement).toBeInTheDocument();
});
