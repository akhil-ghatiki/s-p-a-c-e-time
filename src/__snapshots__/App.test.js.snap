// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="makeStyles-grow-6"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation4"
            style="box-shadow: none;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
            >
              <div
                class="makeStyles-grow-6"
              />
              <div
                class="makeStyles-sectionDesktop-10"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                  style="margin-right: 90px;"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Categories
                    </p>
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                  style="margin-right: 90px;"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Tags
                    </p>
                  </span>
                </button>
                <button
                  aria-controls="primary-search-account-menu"
                  aria-haspopup="true"
                  aria-label="account of current user"
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                  edge="end"
                  style="margin-right: 90px;"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      Contact
                    </p>
                  </span>
                </button>
              </div>
              <div
                class="makeStyles-sectionMobile-11"
              >
                <button
                  aria-controls="primary-search-account-menu-mobile"
                  aria-haspopup="true"
                  aria-label="show more"
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1"
                    >
                      |||
                    </p>
                  </span>
                </button>
              </div>
            </div>
          </header>
        </div>
        <div
          class="MuiGrid-root App-root-1 MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
          >
            <div>
              <div
                class="markdown-body"
              >
                <div
                  class="makeStyles-markdownText-14"
                />
              </div>
            </div>
            <p
              class="App-commentHeader-4"
            >
              Try adding a comment in 
              <span>
                <a
                  href="https://github.com/akhil-ghatiki/akhil-ghatiki.github.io/issues/undefined"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  this issue
                </a>
              </span>
               and it should show up in the comments below
            </p>
            <h6
              class="MuiTypography-root App-commentHeader-4 MuiTypography-h6"
            >
              Comments:
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root"
      id="primary-search-account-menu-mobile"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiMenu-list MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-buttons-12 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Categories
            </p>
          </li>
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-buttons-12 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Tags
            </p>
          </li>
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root makeStyles-buttons-12 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Contact
            </p>
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="makeStyles-grow-6"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation4"
          style="box-shadow: none;"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <div
              class="makeStyles-grow-6"
            />
            <div
              class="makeStyles-sectionDesktop-10"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                style="margin-right: 90px;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Categories
                  </p>
                </span>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                style="margin-right: 90px;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Tags
                  </p>
                </span>
              </button>
              <button
                aria-controls="primary-search-account-menu"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                edge="end"
                style="margin-right: 90px;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Contact
                  </p>
                </span>
              </button>
            </div>
            <div
              class="makeStyles-sectionMobile-11"
            >
              <button
                aria-controls="primary-search-account-menu-mobile"
                aria-haspopup="true"
                aria-label="show more"
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-12 MuiButton-colorInherit"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    |||
                  </p>
                </span>
              </button>
            </div>
          </div>
        </header>
      </div>
      <div
        class="MuiGrid-root App-root-1 MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
        >
          <div>
            <div
              class="markdown-body"
            >
              <div
                class="makeStyles-markdownText-14"
              />
            </div>
          </div>
          <p
            class="App-commentHeader-4"
          >
            Try adding a comment in 
            <span>
              <a
                href="https://github.com/akhil-ghatiki/akhil-ghatiki.github.io/issues/undefined"
                rel="noopener noreferrer"
                target="_blank"
              >
                this issue
              </a>
            </span>
             and it should show up in the comments below
          </p>
          <h6
            class="MuiTypography-root App-commentHeader-4 MuiTypography-h6"
          >
            Comments:
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
